allennlp_models/common/model_card.py

```diff
@@ -0,0 +1,276 @@
+    """
+
+    dataset: Optional[str] = None
+    motivation: Optional[str] = None
+    preprocessing: Optional[str] = None
+
+    def to_dict(self):
+        info = {}
+        for key, val in self.__dict__.items():
+            if val:
+                info["training_" + key] = val
+        return info
+
+
+@dataclass(frozen=True)
+class QuantitativeAnalyses(ModelCardInfo):
+    """
+    This provides disaggregated evaluation of how the
+    model performed based on chosen metrics.
+    """
+
+    unitary_results: Optional[str] = None
+    intersectional_results: Optional[str] = None
+
+
+@dataclass(frozen=True)
+class EthicalConsiderations(ModelCardInfo):
+    """
+    This highlights any ethical considerations to keep
+    in mind when using the model.
+    Eg. Is the model intended to be used for informing
+    decisions on human life? Does it use sensitive data?
+    What kind of risks are possible, and what mitigation
+    strategies were used to address them?
+    """
+
+    ethical_considerations: Optional[str] = None
+
+
+@dataclass(frozen=True)
+class CaveatsAndRecommendations(ModelCardInfo):
+    """
+    This lists any additional concerns. Eg. were any relevant
+    groups not present in the evaluation data?
+    """
+
+    caveats_and_recommendations: Optional[str] = None
+
+
+class ModelCard(ModelCardInfo):
+    """
+    The model card stores the recommended attributes for model reporting
+    as described in the paper [Model Cards for Model Reporting (Mitchell et al, 2019)]
+    (https://arxiv.org/pdf/1810.03993.pdf).
+
+    # Parameters:
+
+    name: str
+        The model's registered name. If `model_class` is not given, this will be used
+        to find any available `Model` registered with this name.
+    model_class: type, optional
+        If given, the `ModelCard` will pull some default information from the class.
+    display_name: str, optional
+        The pretrained model's display name.
+    archive_file: str, optional
+        The location of model's pretrained weights.
+    overrides: Mapping, optional
+        Optional overrides for the model's architecture.
+    model_details: Union[ModelDetails, str], optional
+    intended_use: Union[IntendedUse, str], optional
+    factors: Union[Factors, str], optional
+    metrics: Union[Metrics, str], optional
+    evaluation_data: Union[EvaluationData, str], optional
+    quantitative_analyses: Union[QuantitativeAnalyses, str], optional
+    ethical_considerations: Union[EthicalConsiderations, str], optional
+    caveats_and_recommendations: Union[CaveatsAndRecommendations, str], optional
+
+    Note: For all the fields that are Union[ModelCardInfo, str], a str input will be
+    treated as the first argument of the relevant constructor.
+
+    """
+
+    _storage_location = "https://storage.googleapis.com/allennlp-public-models/"
+
+    def __init__(
+        self,
+        name: str,
+        model_class: Optional[type] = None,
```

Comment:
Yes, I was considering adding a function that loops over the list of available models, so that the model_class can be specified using a string name like `SimpleTagger`. I think Registrable only offers registered_name --> modulepath.ModelType right now.